app/views/layouts/admin.html.erb

```diff
@@ -85,11 +74,8 @@
         </div>
       <% end %>
 
-      <% if FeatureFlag.enabled?(:admin_restructure) %>
-      <!-- hack to ensure that when you go to the old admin view nothing breaks -->
-        <% if request.path.split("/")[-3] == "admin" %>
-          <%= render "admin/shared/tabbed_navbar", menu_items: AdminMenu.nested_menu_items_from_request(request) %>
-        <% end %>
+      <% if request.path.split("/")[-3] == "admin" %>
```

Comment:
It concerns me a little too to be honest and I've been trying to think of ways to improve this as well as the deduced-controller and scope methods. 
However, your suggestion unfortunately won't work as we'll be trying to render the tabbed navbar for any route that has `/admin/` in it instead of the routes that are specifically the admin interface. 

For example: "admin".in?"https://dev.to/t/admin".split("/") will return true and try to render the navbar. Hence I made this condition so specific, to avoid it eagerly trying to render.